core/provider/docker/provider.go

```diff
@@ -86,17 +86,22 @@ func CreateProvider(ctx context.Context, logger *zap.Logger, providerName string
 	}
 
 	dockerProvider.state.NetworkCIDR = cidrMask.String()
+	dockerProvider.state.NetworkGateway = network.IPAM.Config[0].Gateway
 
 	dockerProvider.dockerNetworkAllocator, err = ipallocator.NewCIDRRange(cidrMask)
 
+	if err := dockerProvider.dockerNetworkAllocator.Allocate(net.ParseIP(network.IPAM.Config[0].Gateway)); err != nil {
```

Comment:
do we have a guarantee that `network.IPAM.Config[0]` exists, i.e. that the array is not `nil`?

Reply:
good catch